doc: zms: add calculators

kartben opened the pull request:
Add two HTML calculators embedded in the ZMS documentation:

- Available Space Calculator: computes total data capacity for a given configuration (sector size, sector count, write block size, data size) with support for both NOR flash and RRAM/MRAM devices
- Device Lifetime Calculator: computes expected storage device lifetime based on write frequency, device endurance, and partition configuration

Both calculators use MathJax for formula rendering and include validation for parameters such as sector size alignment.

<img width="553" height="626" alt="image" src="https://github.com/user-attachments/assets/ac3eb9be-2c4a-423f-b12c-b56e2833c4a0" />

This commit was authored with the help of coding agents.

kartben committed:
doc: zms: add interactive calculators for available space & device lifetime

Add two HTML calculators embedded in the ZMS documentation:

- Available Space Calculator: computes total data capacity for a given
  configuration (sector size, sector count, write block size, data size)
  with support for both NOR flash and RRAM/MRAM devices
- Device Lifetime Calculator: computes expected storage device lifetime
  based on write frequency, device endurance, and partition configuration

Both calculators use MathJax for formula rendering and include validation
for parameters such as sector size alignment.

This commit was authored with the help of coding agents.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

rghaddab commented on doc/services/storage/zms/zms.rst:
```diff
@@ -259,6 +259,126 @@ Each key-value pair needs an extra 16 bytes for the ATE, which makes it possible
+            const dataCount = parseInt(document.getElementById('space-data-count').value);
+
+            // Constants
+            const ATE_SIZE = 16;
```
we need to add a round_up function to align all these sizes to the write-block-size.
in C it looks like this :
static inline size_t zms_round_up_write_block_size(size_t len, size_t wite-block-size)
{
	return (len + (write-block-size - 1U)) & ~(write-block-size - 1U);
}

ATE_SIZE = zms_round_up_write_block_size(16, wite-block-size)

rghaddab commented on doc/services/storage/zms/zms.rst:
```diff
@@ -259,6 +259,126 @@ Each key-value pair needs an extra 16 bytes for the ATE, which makes it possible
+        <script>
+        function calculateAvailableSpace() {
+            // Get input values
+            const sectorSize = parseInt(document.getElementById('space-sector-size').value);
```
We should verify that the sectorSize is aligned to write-block-size, otherwise we report an error.
It should be also aligned to the erase-block-size if the device needs an explicit erase.
Which means that we need the user to enter which type of flash is he using.
If he selects flash-type-explicit-erase, he needs to enter as well the erase-block-size, otherwise it is optional.

rghaddab commented on doc/services/storage/zms/zms.rst:
```diff
@@ -332,6 +452,133 @@ Where:
+                </div>
+
+                <div>
+                    <label for="max-writes" style="display: block; margin-bottom: 0.5rem; font-weight: 600; color: var(--admonition-note-color);">Max Writes per Cell:</label>
```
Maybe change the label to be more explicit : Maximum writes per Cell before wear-out

rghaddab commented on doc/services/storage/zms/zms.rst:
```diff
@@ -332,6 +452,133 @@ Where:
+            const dataCount = parseInt(document.getElementById('data-count').value);
+
+            // Constants
+            const ATE_SIZE = 16;
```
Same as above comments

rghaddab commented on doc/services/storage/zms/zms.rst:
```diff
@@ -259,6 +259,126 @@ Each key-value pair needs an extra 16 bytes for the ATE, which makes it possible
+                <div id="space-results-content" style="font-family: var(--monospace-font-family); color: var(--body-color);"></div>
+            </div>
+        </div>
+
```
What do you think about adding the calculations when using ZMS with settings subsystem here as well ?
The calculations of the effective size is a little bit different as for each Settings entry, we need ,in addition to the above calculations, to add ATE_SIZE + average_size_of_path (entry that we should get from user) + ATE_SIZE (for linked list entry). This overhead is for each Settings entry

rghaddab reviewed (commented):
It is very nice to have such graphical content added to the documentation.
Thank you @kartben for having spent some time to generate this.
Here are some initial comments.

kartben commented:
> @kartben I was also doing some changes in a local branch where I updated some calculations and changed the appearance of some fields. Do you prefer that I create a new PR (keeping your signoff) or that you push my changes to your kartben:zms_calc branch ?

If making suggested changes on the PR is too cumbersome then yeah, feel free to create an alternate PR

kartben commented:
@rghaddab I see you removed the stale label - are you expecting me to do anything with this PR? :) You were mentioning you might either do your own PR or make comments here?

rghaddab commented:
> @rghaddab I see you removed the stale label - are you expecting me to do anything with this PR? :) You were mentioning you might either do your own PR or make comments here?

I am taking care of this these days.
I will mention you in the newly opened PR
